src/lib/fsm/index.ts

```diff
@@ -0,0 +1,54 @@
+export default class fsm {
+  private currentState: string;
+  private states: { [id: string]: { [id: string]: string } } = {};
+
+  constructor(initialState: string, transitions: Array<transition>) {
+    this.currentState = initialState;
+
+    transitions.forEach((transition) => this.addTransition(transition));
+  }
+
+  public CurrentState(): string {
+    get: {
+      return this.currentState;
+    }
+  }
+
+  public handle(message: string) {
+    if (this.states[this.currentState] != undefined) {
```

Comment:
This should **never** be undefined. All "initial-state" and all states, must be defined.

This method should just do:
this.currentState = this.transition(this.getCurrentState(), message)